MSCL/source/mscl/MicroStrain/MIP/NMEA/NmeaPacket.h

```diff
@@ -0,0 +1,172 @@
+        //    The message in bytes
+        Message m_message;
+
+        //Variable: m_talkerId
+        //    The talker ID string from the message
+        std::string m_talkerID;
+
+        //Variable: m_sentenceType
+        //    The sentence type from the message
+        std::string m_sentenceType;
+
+    public:
+        //Function: message
+        //    Sets the message bytes
+        //
+        //Parameters:
+        //    bytes - The message as a vector of bytes
+        void message(const Bytes& bytes);
+
+        //Function: talkerId
+        //    Sets the talker ID string
+        void talkerId(std::string id) { m_talkerID = id; }
+
+        //Function: sentenceType
+        //    Sets the sentence type string
+        void sentenceType(std::string type) { m_sentenceType = type; }
+#endif
+
+        //API Function: message
+        //    Gets the message string
+        //
+        //Returns:
+        //    The message as a string
+        const std::string message() const;
+
+        //API Function: message
+        //    Gets the message in bytes
+        //
+        //Returns:
+        //    The message as a vector of bytes
+        const Bytes message_asBytes() const;
+
+        //API Function: talkerId
+        //    Gets the talker ID string
+        //
+        //Returns:
+        //    The talker ID string from the message
+        const std::string talkerId() const { return m_talkerID; }
+
+        //API Function: sentenceTypes
+        //    Gets the sentence type string
+        //
+        //Returns:
+        //    The sentence type string from the message
+        const std::string sentenceType() const { return m_sentenceType; }
+
+        //API Function: size
+        //    Gets the length of the message
+        //
+        //Returns:
+        //    The length of the message
+        const size_t size() const { return m_message.size(); }
+    };
+
+    //API Typedef: NmeaPackets
+    //    Typedef for a vector of <NmeaPacket> objects.
+    typedef std::vector<NmeaPacket> NmeaPackets;
+
+#ifndef SWIG
+    //Struct: NmeaPacketInfo
+    //    Contains constants representing information on NMEA Packets
+    struct NmeaPacketInfo
+    {
+        //=====================================================================================================
+        //Enum: NMEA identifiers
+        //    NMEA_START_OF_PACKET      - "$"       - The start of a NMEA packet/sentence
+        //    NMEA_START_OF_PACKET_ALT  - "!"       - The start of a NMEA encapsulation sentence
+        //    NMEA_END_OF_PACKET        - "\r\n"    - The end of a NMEA packet/sentence
+        //    NMEA_DATA_START_DELIM     - ","       - Characters before first instance is header info, ',' delimiter used throughout message
+        //    NMEA_CHECKSUM_DELIM       - "*"       - NMEA checksum delimiter
+        //=====================================================================================================
+        enum
```

Comment:
I feel like a `static constexpr` or `static const` would be better than an enum here since these variables are of different types